src/pat/inject.js

```diff
@@ -72,7 +72,7 @@ define([
                         $el.on("click.pat-inject", _.onClick);
                     } else if ($el.is("form")) {
                         $el.on("submit.pat-inject", _.onSubmit)
-                        .on("click.pat-inject", "[type=submit]", ajax.onClickSubmit);
+                        .on("click.pat-inject", "[type=submit]", _.onButtonClick);
```

Comment:
You can differentiate between formaction and non-formaction use-cases here by using different jQuery selectors.

For example:
```
$el.on("submit.pat-inject", _.onSubmit)
    .on("click.pat-inject", "[type=submit]", ajax.onClickSubmit)
    .on("click.pat-inject", "[type=submit][formaction]", _.onButtonClick);
```

The ``onButtonClick`` method can then be simpler and smaller because it doesn't need to do the ``if (formaction)`` check. I would then also rename ``onButtonClick`` to something like ``onFormActionSubmit``.

The ``formaction`` attribute can also be used on buttons of ``type="image"`  (see [here](http://html5doctor.com/html5-forms-introduction-and-new-attributes/#formaction)). So the selector needs to be extended.

So, sticking with the previous example, it would look like this:

```
$el.on("submit.pat-inject", _.onSubmit)
    .on("click.pat-inject", "[type=submit]", ajax.onClickSubmit)
    .on("click.pat-inject", "[type=submit][formaction], [type=image][formaction]", _.onButtonClick);
```